### ISSUE TITLE
' should not be a pair in haskell mode.

### ISSUE THREAD

**Issue by IvanMalison** [Fuco1/smartparens#710]
Apostrophies are often used in identifiers in haskell to indicate that something is a modified version of another thing.

Smart parens has a bunch of annoying behavior with apostrophies in haskell mode

Here is an example:

``` haskell
module Main where

myFunction a = a + 1

myFunction' 1 = 1
myFunction' a = myFunction a + 1

test c d = myFunction' c + (myFunction' d)

main = return ()
```

Running sp-splice-sexp inside of the uncessary parentheses on the test line has very suprising behavior.

EDIT:

Just noticed that smartparens-haskell has 
```
(sp-local-pair "'" nil :unless '(sp-point-after-word-p))
```

Which I suspect might be why this still occurs. What is the reasoning behind this piece of config?

**Comment by Fuco1** [comment]
The config line should actually do what you want, it should disable the pair when it is at the end of the token (suck as function name with a ' at the end).  However, the problem seems to be that this is not ignored in the parsing routines.

This should be somewhat simple to fix, I'll try to get it working.

**Comment by Fuco1** [comment]
Well, this was a combination of some weird bugs I've uncovered while trying to fix the haskell setup.  So those are also fixed :) I've added some tests covering this report so it shouldn't repeat in the future.

**Comment by IvanMalison** [comment]
Cool. So it's fixed?

**Comment by Fuco1** [comment]
It is now.  Will be released in 1.11 but you can get it from MELPA probably in couple hours.
